### PR TITLE
repair pegging in thread "TerminalBoard control keeper"

### LINKED ISSUE #8: Thread "TerminalBoard control keeper" is pegging
Not sure it was always the case, but right from the start the thread "TerminalBoard control keeper" is pegging a CPU core.

### PR DESCRIPTION
**Pull request by Ra-V2**
Fixes #8 
The problem here was loop while(true)

### CODE REVIEW

File: TerminalBoard.cpp [hunk 1]
```diff
@@ -71,6 +71,7 @@ control(void *data)
 	BMessenger messenger(NULL, target);
 	while (true)
 	{
+		usleep(100);
 		int32 c = std::cin.get();
```
Comment (pulkomandy) [inline TerminalBoard.cpp]: This should block on the cin here and wait for keyboard input.
But, this is not the right way to handle keyboard input in a GUI application. Instead the KeyDown method in BView should be used.

**Comment by Ra-V2** [comment]
From what I understood, is it supposed to be something like this?

File: TerminalBoard.cpp [hunk 2]
```diff
@@ -132,3 +102,21 @@ TerminalBoard::showBoard()
...
 	std::cout << "Score: " << fTarget->Score() << std::endl;
 }
+
+void
+TerminalBoard::KeyDown(const char *bytes, int32 numBytes)
+{
+	BMessenger messenger(NULL, fTarget);
+	if ( numBytes == 1 ) {
+		int32 c = bytes[0];
+		if (c == 'w' || c == 'a' || c == 's' || c == 'd'){
```
Comment (mmuman) [inline TerminalBoard.cpp]: You probably also want to check for `B_UP_ARROW` and friends so people can use the arrow keys, see `headers/os/interface/InterfaceDefs.h`.

File: TerminalBoard.cpp [hunk 3]
```diff
@@ -132,3 +102,21 @@ TerminalBoard::showBoard()
...
+			BMessage move(H2048_MAKE_MOVE);
+			move.AddInt32("direction", c);
+			messenger.SendMessage(&move);
+		} else if (c == 'u') {
```
Comment (mmuman) [inline TerminalBoard.cpp]: Maybe also `B_BACKSPACE` ?

File: TerminalBoard.cpp (hunk 1 above)
Comment (mmuman) [reply TerminalBoard.cpp]: Wait, isn't this "TerminalBoard"? This looks like a terminal interface, the GUI one is in WindowBoard.

Comment (mmuman) [reply TerminalBoard.cpp]: In which case, the proper way would be to use select() on the stdin descriptor or something alike.

**Comment by Ra-V2** [comment]
I removed the wsad control because it is not implemented in the game.

**Comment by mmuman** [comment]
Looks not too bad. You'll want to squash your commits into a single one to get rid of the other attempts.

**Comment by pulkomandy** [comment]
Does it actually work to use B_UP_ARROW, etc in this context? They are not meant for use in this context and do not match the characters that the arrows would return. Please test your code: run 2048 in a Terminal and try to use the arrows (or the letters, as it was before) to play there (without using the UI window).

Still, I wonder why we need this feature at all, it seems useless to play in the Terminal in this app? Couldn't we just remove all this code?

**Comment by Ra-V2** [comment]
> Does it actually work to use B_UP_ARROW, etc in this context? They are not meant for use in this context and do not match the characters that the arrows would return. Please test your code: run 2048 in a Terminal and try to use the arrows (or the letters, as it was before) to play there (without using the UI window).
> 
> Still, I wonder why we need this feature at all, it seems useless to play in the Terminal in this app? Couldn't we just remove all this code?

Arrow keys doesn't work but the KeyDown function works pretty good. I also think that nobody will play this game in the terminal.

**Comment by pulkomandy** [comment]
Yes, so let's just remove that TerminalBoard code completely, it's useless.

**Comment by mmuman** [comment]
Indeed, it does fix #8 .

Can you please rebase and squash your commits?
Provided you cloned from HaikuArchive (which creates the remote `origin`) and added your fork afterward, you can do it with `git rebase -i origin` and change all lines after the first one from `pick` to `squash`.

This will make a single commit. If you want to edit the commit log then, you can `git commit --amend`.

Then you can `git push -f github` (or whatever remove you named your fork).

**Comment by Ra-V2** [comment]
> Indeed, it does fix #8 .
> 
> Can you please rebase and squash your commits?
> Provided you cloned from HaikuArchive (which creates the remote `origin`) and added your fork afterward, you can do it with `git rebase -i origin` and change all lines after the first one from `pick` to `squash`.
> 
> This will make a single commit. If you want to edit the commit log then, you can `git commit --amend`.
> 
> Then you can `git push -f github` (or whatever remove you named your fork).

Done.

**Comment by mmuman** [comment]
Thanks!

**Pull request merged**